File: server/data/results/translations/lite/plain/java/fibonacci_memo.java
import java.util.HashMap;
import java.util.Map;

public class Main {
    public static long fib(int n, Map<Integer, Long> memo) {
        if (n <= 1) {
            return n;
        }
        if (memo.containsKey(n)) {
            return memo.get(n);
        }
        long value = fib(n - 1, memo) + fib(n - 2, memo);
        memo.put(n, value);
        return value;
    }

    public static void main(String[] args) {
        Map<Integer, Long> memo = new HashMap<>();
        for (int i = 0; i < 15; i++) {
            System.out.println(fib(i, memo));
        }
    }
}

File: server/data/translation_bench/fibonacci_memo.py
def fib(n, memo):
    if n <= 1:
        return n
    if n in memo:
        return memo[n]
    value = fib(n - 1, memo) + fib(n - 2, memo)
    memo[n] = value
    return value


def main():
    memo = {}
    for i in range(0, 15):
        print(fib(i, memo))


main()
